src/modules/animal/index.ts

```diff
@@ -17,6 +17,41 @@ export class AnimalModule {
     }
   }
 
+  /**
+   * return randomly one of animals different types.
+   *
+   * @example
+   * faker.animal.random() // 'Sloth bear'
+   *
+   * @since 8.0.0
+   */
+  random(): string {
+    // define array of animals
+    const typeOfAnimals = [
+      'bear',
+      'bird',
+      'cat',
+      'cetacean',
+      'cow',
+      'crocodilia',
+      'dog',
+      'fish',
+      'horse',
+      'insect',
+      'lion',
+      'rabbit',
+      'rodent',
+      'snake',
+    ];
+
+    // randomly select  one of typeOfAnimals item
+    const choice: string = this.faker.helpers.arrayElement(typeOfAnimals);
+    //define variable for store result of choice
+    const arrayOfCreatures: string[] = this.faker.definitions.animal[choice];
+    // pick up randomly one of selected animal array
+    return this.faker.helpers.arrayElement(arrayOfCreatures);
+  }
```

Comment:
I think this can be simplified

```suggestion
  /**
   * Returns a random animal.
   *
   * @example
   * faker.animal.sample() // 'Sloth bear'
   *
   * @since 8.0.0
   */
  sample(): string {
    return this.faker.helpers.objectValue(this.faker.definitions.animal);
  }
```

Reply:
I'm not sure about that.

1. The category.entries are string[] and thus objectValue would return an array and not a simple string thereof.
2. I'm not sure whether we implemented the key set method in our definitions proxy.

Reply:
I didn't tested it indeed, but `objectValue` seems to _correctly_ return `T[keyof T]`

https://github.com/faker-js/faker/blob/11a5f51ddb4ab1163d770c4b12f5fe4711b8e11d/src/modules/helpers/index.ts#L474

Reply:
Oh! You mean we need to use `this.faker.helpers.arrayElement(this.faker.helpers.objectValue(this.faker.definitions.animal))`